### PR TITLE
Various fixes for the install scripts

### PR DESCRIPTION
**Pull request by deb0ch**
Following discussion in PX4/px4_ros_com#33

### CODE REVIEW

File: scripts/setup_system.bash [hunk 1]
```diff
@@ -122,25 +122,25 @@ unset ROS_DISTRO
...
+    && sudo apt-get clean autoclean \
     && sudo rm /tmp/python3-genmsg_0.5.11-2_all.deb
 
 # Install python3-gencpp or download and install from deb source (currently only available in Ubuntu 18.10 and above)
-sudo apt-get install -y --quiet python3-gencpp
-  || wget http://mirrors.kernel.org/ubuntu/pool/universe/r/ros-gencpp/python3-gencpp_0.6.0-4_all.deb -P /tmp/
-    && sudo dpkg -i /tmp/python3-gencpp_0.6.0-4_all.deb
-    && sudo apt-get -y autoremove
-    && sudo apt-get clean autoclean
+sudo apt-get install -y --quiet python3-gencpp \
```
Comment (TSC21) [inline scripts/setup_system.bash]: Actually `gencpp` is not actually need at all. Can you please remove it? Thanks!

Comment (deb0ch) [reply scripts/setup_system.bash]: I still got build errors specifically complaining that `gencpp` was missing though, so even if it is not used it is still a dependency for now  ¯\\_(ツ)\_/¯

Comment (deb0ch) [reply scripts/setup_system.bash]: Building `px4_ros_com` using `colcon` with ROS2 Eloquent

Comment (TSC21) [reply scripts/setup_system.bash]: > I still got build errors specifically complaining that `gencpp` was missing though, so even if it is not used it is still a dependency for now ¯\_(ツ)_/¯

Yeah, https://github.com/PX4/Firmware/pull/13572 will remove the need of it, but there's no need to keep it in the script.

> Building `px4_ros_com` using `colcon` with ROS2 Eloquent

@deb0ch again, if you want to see this working, you will have to use Crystal.

**Comment by deb0ch** [comment]
No problem ! 👌✨ 

@deb0ch again, if you want to see this working, you will have to use Crystal.
Yeah, I just browsed the discussion in eProsima/Fast-RTPS#829 and the problem looks pretty hairy. Do you think I could build my own ROS2 distribution from source, mixing package versions between Crystal, Dashing and Eloquent, to get this working and still benefit from the large amount of features that Dashing and Eloquent brought in ?

**Comment by TSC21** [comment]
> Do you think I could build my own ROS2 distribution from source, mixing package versions between Crystal, Dashing and Eloquent, to get this working and still benefit from the large amount of features that Dashing and Eloquent brought in ?

From my experience, that's not a good idea and you will bump into a lot of issues where depencies are not met, or large API changes result in incompatible code, etc etc. My advice: wait a bit more and I should find my way into solving this - I am quite close actually.

**Comment by TSC21** [comment]
@deb0ch CI seems to be failing. I am not sure if it's because of your changes or something else.

**Comment by deb0ch** [comment]
Yes, I am trying to figure that out 😅 

There is a `/bin/bash: debug_vect_advertiser: command not found` at the end of the logs, somewhere a script fails with `PermissionError: [Errno 13] Permission denied: 'log'` at line 625 of the logs.

**Comment by deb0ch** [comment]
`micrortps_agent` is missing, some other program as well, access rights error on `log`, and the `build_ros2_workspace.bash` doesn't attempt to install any of these, so I would say it could be related with the setup of the containers 🤔

**Comment by deb0ch** [comment]
I do not think that this PR caused the CI to fail: the only dependency that I changed is `build_ros2_workspace.bash`, only the `SCRIPT_DIR` variable has changed and its value remains the same in my testings.

**Comment by TSC21** [comment]
> `micrortps_agent` is missing, some other program as well, access rights error on `log`, and the `build_ros2_workspace.bash` doesn't attempt to install any of these, so I would say it could be related with the setup of the containers

The containers were lastly updated the beginning of November, and there were successful builds after that in December. Considering those didn't change, I would have to say that this needs to be something in `build_ros2_workspace.bash`.

**Comment by TSC21** [comment]
@deb0ch the line https://travis-ci.com/PX4/px4_ros_com/jobs/272745134#L626 means it is failing while running `build_ros2_workspace.bash`.

**Comment by deb0ch** [comment]
Haha, got the bug 😸:bug:

I didn't account for symlinks, or even that for a strange reason that I don't understand `$(realpath -s .)` would not have the same value as `$PWD` 🤔

**Comment by TSC21** [comment]
> Haha, got the bug 
> 
> I didn't account for symlinks, or even that for a strange reason that I don't understand `$(realpath -s .)` would not have the same value as `$PWD` (only in Travis' machine)

Great! Now I am more than happy to merge! Thanks!

**Comment by deb0ch** [comment]
Yay ! \o/

**Comment by deb0ch** [comment]
I am using this conversation as I don't know where to post this without polluting a discussion, as there is no place for informal chat-like discussion in Github. Do you know one for these kinds of subjects ?

I stumbled upon this browsing the docs: https://index.ros.org/doc/ros2/Releases/Release-Dashing-Diademata/#rosidl Reading your issues I thought it might be related to your idl problem porting px4_ros_com to dashing. I am sure you already solved that part of your problem and the soution doesn't seem that simple, but you never know what might help  ¯\\_(ツ)\_/¯ 

On another note, are you able to estimate when PX4 will be available in Dashing, and from there, how long or difficult it might be to update to Eloquent ?

For context, I would like to use MoveIt! 2 for its 3D perception and path planning features along with with PX4 for my current project (I know it is only available in Dashing, I am asking about Eloquent out of curiosity). If you happen to have some insights that you would be willing to share I would be more than happy 😸

**Comment by deb0ch** [comment]
@TSC21 I am using this conversation as I don't know where to post this without polluting a discussion, as there is no place for informal chat-like discussion in Github. Do you know one for these kinds of subjects ?

I stumbled upon this browsing the docs: https://index.ros.org/doc/ros2/Releases/Release-Dashing-Diademata/#rosidl Reading your issues I thought it might be related to your idl problem porting px4_ros_com to dashing. I am sure you already solved that part of your problem and the soution doesn't seem that simple, but you never know what might help  ¯\\_(ツ)\_/¯ 

On another note, are you able to estimate when PX4 will be available in Dashing, and from there, how long or difficult it might be to update to Eloquent ?

For context, I would like to use MoveIt! 2 for its 3D perception and path planning features along with with PX4 for my current project (I know it is only available in Dashing, I am asking about Eloquent out of curiosity). If you happen to have some insights that you would be willing to share I would be more than happy 😸

**Comment by TSC21** [comment]
> @TSC21 I am using this conversation as I don't know where to post this without polluting a discussion, as there is no place for informal chat-like discussion in Github. Do you know one for these kinds of subjects ?

Did you read the repository `README`? :)

**Comment by TSC21** [comment]
> @TSC21 I am using this conversation as I don't know where to post this without polluting a discussion, as there is no place for informal chat-like discussion in Github. Do you know one for these kinds of subjects ?

I can't because it's not completely dependent of my technical capabilities at this point :) Sorry. As soon as a solution is found I will let you know.

**Comment by deb0ch** [comment]
> Did you read the repository README? :) https://github.com/PX4/px4_ros_com#questions-and-troubleshooting

Haha, it seems like I managed to miss this section 😅 Thank you !

**Pull request merged**